src/cinder/gl/GlslProg.cpp

```diff
@@ -697,10 +718,20 @@ void GlslProg::loadShader( const string &shaderSource, const fs::path &shaderPat
 			}
 			log += "\n" + ss.str();
 		}
-#endif		
+#endif
+		// Since the GlslProg destructor will not be called after we throw, we must delete all
+		// owned GL objects here to avoid leaking. Any other attached shaders will be cleaned up
+		// when the program is deleted
+		glDeleteShader( handle );
+		glDeleteProgram( mHandle );
+
 		throw GlslProgCompileExc( log, shaderType );
 	}
+
 	glAttachShader( mHandle, handle );
+	glDeleteShader( handle );
```

Comment:
Moving this to a utility seems fine. Also, I'd kind of like to get those ternary operators out of the constructor.. How would you feel about checking the source string for `.empty()` int `loadShader()` and just returning 0 there? Seems cleaner.